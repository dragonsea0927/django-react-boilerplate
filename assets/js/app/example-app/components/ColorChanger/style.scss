.main-container {
  border: 1px dashed grey;
  margin: 5px;
  padding: 5px;

  .app-name {
    color: red;
    padding-top: 20px;
  }

  .inner-container {
    margin: 0;
    padding: 5px;

    .color-picker {
      margin: 0px 5px 0px 5px;
    }
  }
}
