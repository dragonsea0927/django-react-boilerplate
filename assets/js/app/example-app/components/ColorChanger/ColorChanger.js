import React from 'react';
import PropTypes from 'prop-types';

import ColorDisplay from '../ColorDisplay';

import './style.scss';


class ColorChanger extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      color: 'black',
    };

    this.handleChangeColor = this.handleChangeColor.bind(this);
  }

  handleChangeColor(e) {
    const color = e.target.value;
    this.setState({ color });
  }

  render() {
    const { title } = this.props;

    return (
      <div className="main-container">
        <h2>{title}</h2>
        <h3 className="app-name">Color Changer App</h3>
        <p>
          Check this example app: change the color to see it reflected in the text next to it.
        </p>

        <div className="inner-container">
          <select className="color-picker" onChange={this.handleChangeColor}>
            <option value="black">Black</option>
            <option value="green">Green</option>
            <option value="red">Red</option>
            <option value="blue">Blue</option>
            <option value="purple">Purple</option>
          </select>

          <ColorDisplay color={this.state.color} />
        </div>
      </div>
    );
  }
}

ColorChanger.defaultProps = {
  title: 'React App Loaded!',
};

ColorChanger.propTypes = {
  title: PropTypes.string,
};

export default ColorChanger;
