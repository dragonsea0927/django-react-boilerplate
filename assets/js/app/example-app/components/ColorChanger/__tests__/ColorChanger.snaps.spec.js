import React from 'react';
import renderer from 'react-test-renderer';

import ColorChanger from '../ColorChanger';

jest.mock('../../ColorDisplay/ColorDisplay');


describe('ColorChanger', () => {
  let Component;
  let tree;

  test('Some title', () => {
    Component = renderer.create(
      <ColorChanger title="This is a test title" />);

    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('No title (should use default)', () => {
    Component = renderer.create(
      <ColorChanger />);

    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
