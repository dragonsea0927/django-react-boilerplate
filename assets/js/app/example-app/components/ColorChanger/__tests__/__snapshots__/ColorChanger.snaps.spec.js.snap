// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorChanger No title (should use default) 1`] = `
<div
  className="main-container"
>
  <h2>
    React App Loaded!
  </h2>
  <h3
    className="app-name"
  >
    Color Changer App
  </h3>
  <p>
    Check this example app: change the color to see it reflected in the text next to it.
  </p>
  <div
    className="inner-container"
  >
    <select
      className="color-picker"
      onChange={[Function]}
    >
      <option
        value="black"
      >
        Black
      </option>
      <option
        value="green"
      >
        Green
      </option>
      <option
        value="red"
      >
        Red
      </option>
      <option
        value="blue"
      >
        Blue
      </option>
      <option
        value="purple"
      >
        Purple
      </option>
    </select>
    <span>
      ColorDisplay 
      {"color":"black"}
    </span>
  </div>
</div>
`;

exports[`ColorChanger Some title 1`] = `
<div
  className="main-container"
>
  <h2>
    This is a test title
  </h2>
  <h3
    className="app-name"
  >
    Color Changer App
  </h3>
  <p>
    Check this example app: change the color to see it reflected in the text next to it.
  </p>
  <div
    className="inner-container"
  >
    <select
      className="color-picker"
      onChange={[Function]}
    >
      <option
        value="black"
      >
        Black
      </option>
      <option
        value="green"
      >
        Green
      </option>
      <option
        value="red"
      >
        Red
      </option>
      <option
        value="blue"
      >
        Blue
      </option>
      <option
        value="purple"
      >
        Purple
      </option>
    </select>
    <span>
      ColorDisplay 
      {"color":"black"}
    </span>
  </div>
</div>
`;
